src/login/login_screen.rs

```diff
@@ -329,5 +484,12 @@ pub enum LoginAction {
         password: String,
         homeserver: Option<String>,
     },
+    /// True refers to await user's SSO login in a browser. This check primary prevents
+    /// spawning another sso http server  
+    SsoPending(bool),
+    /// A list of identity providers from matrix login_type api. 
+    /// As not all homeservers have the same identity providers, 
+    /// this determines the number of different sso login types in the frontend
+    IdentityProvider(Vec<IdentityProvider>),
```

Comment:
Here's an example of the kind of comments we're looking for. The first line is a summary of the item, and then additional lines are separated by a new line and then followed by additional details. If you check how rustdoc formats these doc comments, you'll see _why_ we format it in that way.
```suggestion
    /// An acknowledgment that is sent from the backend Matrix task to the login screen
    /// informing it that the SSO login process is either still in flight (`true`) or has finished (`false`).
    ///
    /// Note that an inner value of `false` does *not* imply that the login request has
    /// successfully finished. 
    /// The login screen can use this to prevent the user from submitting
    /// additional SSO login requests while a previous request is in flight. 
    SsoPending(bool),
    /// A list of SSO identity providers supported by the homeserver.
    ///
    /// This is sent from the backend async task to the login screen in order to
    /// inform the login screen which SSO identity providers it should display to the user.
    IdentityProvider(Vec<IdentityProvider>),
```